AppBuilder/platform/ABDataCollection.js

```diff
@@ -319,33 +276,18 @@ module.exports = class ABDataCollection extends ABDataCollectionCore {
     * state.
     * @return {Promise}
     */
-   platformReset() {
-      return new Promise((resolve, reject) => {
-         // Make sure our ABObject is properly setup on the platform
-         this.datasource
-            .model()
-            .local()
-            .platformReset()
-            .then(() => {
-               // once that is done, make sure we can clear our DC info
-               const storage = this.AB.storage;
-               var lock = storage.Lock(this.refStorage());
-               return lock
-                  .acquire()
-                  .then(() => {
-                     return storage.set(this.refStorage(), null);
-                  })
-                  .then(() => {
-                     lock.release();
-                  });
-            })
-            .then(() => {
-               // now clear all our live values:
-               this.clearAll();
-               resolve();
-            })
-            .catch(reject);
-      });
+   async platformReset() {
+      await this.datasource.model().local().platformReset();
+
+      // once that is done, make sure we can clear our DC info
+      const storage = this.AB.storage;
+      const lock = storage.Lock(this.refStorage());
+      await lock.acquire();
+      await storage.set(this.refStorage(), null);
```

Comment:
Isn't this already done in AB model local? Is this needed, or just an extra call to storage?
I have been generally trying to keep logic for accessing `local` data in `model.local`, but it's actually hard to justify in this case since it's such a simple action ---- that is only used here. What do you think?

https://vscode.dev/github/appdevdesigns/appbuilder_platform_mobile/blob/is/update/AppBuilder/platform/ABModelLocal.js#L67